rust_src/src/crypto/mod.rs

```diff
@@ -502,7 +502,7 @@ fn buffer_hash(buffer_or_name: LispObject) -> LispObject {
     if buffer.is_nil() {
         unsafe { nsberror(buffer_or_name.to_raw()) };
     }
-    let b = buffer.as_buffer().unwrap();
+    let mut b = buffer.as_buffer().unwrap();
```

Comment:
This isn't necessary anymore.

Unfortunately this file uses `#![allow(unused)]` which suppresses "unused mut" warnings too.

It should use `#![allow(dead_code)]` instead.